crates/pathfinder/src/sync/error.rs

```diff
@@ -25,3 +27,63 @@ pub(super) enum SyncError {
     #[error("Event commitment mismatch")]
     EventCommitmentMismatch(PeerId),
 }
+
+impl PartialEq for SyncError {
+    fn eq(&self, other: &Self) -> bool {
+        todo!();
+    }
+}
+
+#[derive(Debug, thiserror::Error, Clone)]
+pub(super) enum SyncError2 {
```

Comment:
Same as `SyncError` but without `PeerId` which are instead moved to `SyncResult` as wrappers. This is also non-ideal, there are cases where PeerId is annoying.